File: backend/src/main/java/com/cysz/minimal/enums/UnitPurpose.java
package com.cysz.minimal.enums;

/**
 * 单元用途枚举
 */
public enum UnitPurpose {
    
    OFFICE("OFFICE", "办公"),
    WAREHOUSE("WAREHOUSE", "仓库"),
    FREIGHT("FREIGHT", "货运"),
    COMMERCIAL("COMMERCIAL", "商业"),
    MEETING_ROOM("MEETING_ROOM", "会议室"),
    BUSINESS_ROOM("BUSINESS_ROOM", "营业房"),
    PARKING("PARKING", "停车位"),
    ADVERTISING("ADVERTISING", "广告位"),
    APARTMENT("APARTMENT", "公寓"),
    MULTI_BUSINESS("MULTI_BUSINESS", "多经点位"),
    PROMOTION_VENUE("PROMOTION_VENUE", "推广场地");
    
    private final String code;
    private final String displayName;
    
    UnitPurpose(String code, String displayName) {
        this.code = code;
        this.displayName = displayName;
    }
    
    public String getCode() {
        return code;
    }
    
    public String getDisplayName() {
        return displayName;
    }
    
    /**
     * 根据代码获取枚举
     */
    public static UnitPurpose fromCode(String code) {
        if (code == null) {
            return null;
        }
        for (UnitPurpose purpose : UnitPurpose.values()) {
            if (purpose.getCode().equals(code)) {
                return purpose;
            }
        }
        throw new IllegalArgumentException("未知的单元用途代码: " + code);
    }
    
    /**
     * 根据代码获取显示名称
     */
    public static String getDisplayNameByCode(String code) {
        UnitPurpose purpose = fromCode(code);
        return purpose != null ? purpose.getDisplayName() : code;
    }
    
    /**
     * 检查代码是否有效
     */
    public static boolean isValidCode(String code) {
        if (code == null) {
            return false;
        }
        for (UnitPurpose purpose : UnitPurpose.values()) {
            if (purpose.getCode().equals(code)) {
                return true;
            }
        }
        return false;
    }
}

File: frontend/src/constants/unitPurpose.ts
/**
 * 单元用途相关常量和工具函数
 */

// 单元用途选项
export const UNIT_PURPOSE_OPTIONS = [
  { value: 'OFFICE', label: '办公' },
  { value: 'WAREHOUSE', label: '仓库' },
  { value: 'FREIGHT', label: '货运' },
  { value: 'COMMERCIAL', label: '商业' },
  { value: 'MEETING_ROOM', label: '会议室' },
  { value: 'BUSINESS_ROOM', label: '营业房' },
  { value: 'PARKING', label: '停车位' },
  { value: 'ADVERTISING', label: '广告位' },
  { value: 'APARTMENT', label: '公寓' },
  { value: 'MULTI_BUSINESS', label: '多经点位' },
  { value: 'PROMOTION_VENUE', label: '推广场地' }
] as const

// 单元用途枚举类型
export type UnitPurpose = typeof UNIT_PURPOSE_OPTIONS[number]['value']

// 单元用途映射对象
export const UNIT_PURPOSE_MAP = Object.fromEntries(
  UNIT_PURPOSE_OPTIONS.map(item => [item.value, item.label])
) as Record<UnitPurpose, string>

/**
 * 根据用途代码获取显示名称
 * @param purpose 用途代码
 * @returns 显示名称
 */
export const getUnitPurposeName = (purpose: string | null | undefined): string => {
  if (!purpose) return '-'
  return UNIT_PURPOSE_MAP[purpose as UnitPurpose] || purpose
}

/**
 * 检查用途代码是否有效
 * @param purpose 用途代码
 * @returns 是否有效
 */
export const isValidUnitPurpose = (purpose: string): purpose is UnitPurpose => {
  return UNIT_PURPOSE_OPTIONS.some(option => option.value === purpose)
}